python_modules/dagster/dagster/_core/storage/legacy_storage.py

```diff
@@ -463,9 +463,11 @@ def wipe_asset(self, asset_key: "AssetKey"):
         return self._storage.event_log_storage.wipe_asset(asset_key)
 
     def get_materialization_count_by_partition(
-        self, asset_keys: Sequence["AssetKey"]
+        self, asset_keys: Sequence["AssetKey"], after_cursor: Optional[int] = None
```

Comment:
Consider if we should just apply a generic `EventRecordsFilter` in here... might simplify the signature a bit, with the tradeoff that it could be confusing if some conflicting args were applied (e.g. non materialization event types).

Reply:
I did think on this for awhile, as there really are plenty of useful fields on the EventRecordsFilter that you could hypothetically want to pass in here (`partition_keys`, `before_cursor`, and `tags` could all be useful in theory)  but I think the confusion of conflicting args might be too much -- EventRecordsFilter doesn't (yet?) support an `asset_keys` argument (just `asset_key`), and so there'd be two gotchas (you can't set the `asset_key` argument, and you need to set the `event_type` to `DagsterEventType.ASSET_MATERIALIZATION`).